packages/react-dom/src/__tests__/ReactTestUtilsAct-test.js

```diff
@@ -0,0 +1,326 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ */
+
+let React;
+let ReactDOM;
+let ReactTestUtils;
+let act;
+
+jest.useRealTimers();
+
+function sleep(period) {
+  return new Promise(resolve => {
+    setTimeout(() => {
+      resolve(true);
+    }, period);
+  });
+}
+
+describe('ReactTestUtils.act()', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    React = require('react');
+    ReactDOM = require('react-dom');
+    ReactTestUtils = require('react-dom/test-utils');
+    act = ReactTestUtils.act;
+  });
+
+  describe('sync', () => {
+    it('can use act to flush effects', () => {
+      function App(props) {
+        React.useEffect(props.callback);
+        return null;
+      }
+
+      let called = false;
+      act(() => {
+        ReactDOM.render(
+          <App
+            callback={() => {
+              called = true;
```

Comment:
This seems nitpicky but by convention in our tests we push things into a log and then assert on the log. That way if, say, the callback is called twice instead of once, the test will fail.

Reply:
Good point, I’ll make it so

Reply:
I fixed this, but I found a bunch of other usages of this pattern in the tests haha https://github.com/facebook/react/search?q=%22called+%3D+true%22&type=Code

Reply:
Now we have fewer! :D